tsfc/fem.py

```diff
@@ -383,18 +145,17 @@ def __getitem__(self, key):
 class Translator(MultiFunction, ModifiedTerminalMixin, ufl2gem.Mixin):
     """Contains all the context necessary to translate UFL into GEM."""
 
-    def __init__(self, weights, quadrature_index, argument_indices, tabulation_manager,
-                 coefficient_map, index_cache):
+    def __init__(self, tabulation_manager, weights, quadrature_index, coefficient_mapper, index_cache):
         MultiFunction.__init__(self)
         ufl2gem.Mixin.__init__(self)
         integral_type = tabulation_manager.integral_type
         self.weights = gem.Literal(weights)
         self.quadrature_index = quadrature_index
-        self.argument_indices = argument_indices
         self.tabulation_manager = tabulation_manager
         self.integral_type = integral_type
-        self.coefficient_map = coefficient_map
+        self.coefficient_mapper = coefficient_mapper
         self.index_cache = index_cache
+        self.argument_indices = (gem.Index(name='j'), gem.Index(name='k'))
```

Comment:
I have changed my mind again. I think the current version is less convoluted than both the original and the first proposed.